### DISCUSSION TITLE
Add option to enter time in DatePicker component

### DISCUSSION THREAD

**Discussion by dukesx** [mantinedev/mantine#1058] [Feature requests]
### Proposal
Date Time Picker component, a component that will make it hell of alot easy to cherry pick not only month, not only day, but also the time of the day you want. 

#### What's the use ? 
Have you ever booked an appointment with a clinic or a doctor ? there you have it, biggest and most common use of it.

#### Why not combine existing Individual components and make it ?
Ask yourself, would you prefer the hassle of individually maintaining the components  along with state or just have the ease of having both functions, under same component, styled according to guidelines ? 

#### Any Example/Design Spec ?
![date-time](https://user-images.githubusercontent.com/5988991/131878447-0fd02eda-41da-4633-83e7-0047432eea4e.png)

Ant design of course, A very mature example.

**Comment by dukesx** [comment]
@rtivital there you go !

**Comment by rtivital** [comment]
Okay, thanks, if everyone is interested in implementing this, ping me here and I'll provide instructions

**Comment by UltraInstinct05** [comment]
I'll try to work on this one. I believe this is the relevant trello card to track it ? https://trello.com/c/1nKqmma2/39-mantine-dates-datepicker-add-optional-timeinput (I guess GitHub project board is better, everything is kept in one place)

**Comment by rtivital** [comment]
Okay, sounds good

**Comment by mattrunyon** [comment]
Any updates on this? Might be able to take a stab at it

**Comment by kiboo-eth** [comment]
I would love this feature

**Comment by cybul26** [comment]
I would really appreciate this feature. Mantine has many great components and I'm suprised DateTimePicker is missing.

**Comment by skelawsky** [comment]
+1 even basic time picker (like standard `<input type='time' />`) would be super good feature.

**Comment by nammi-kobler** [comment]
My team would also really love this feature! It sucks to keep MUI just for a combined Date and Time picker!

**Comment by manueltollis** [comment]
This would be such a usefull feature to have! I'm working right now on a  web app based on creating events and this would be so perfect!

**Comment by mjkuhlen** [comment]
This would definitely be a great feature to have.

**Comment by horsebnb** [comment]
Guys in the mean time, you can just use the DatePicker and TimeInput component, and then add them together with this function:

```
import dayjs from "dayjs";

export default function combineTimeAndDate(time, date) {
  if (!(date instanceof Date)) return undefined;
  if (!(time instanceof Date)) return date;

  const hour = dayjs(time).hour();
  const minute = dayjs(time).minute();
  const dateAndTime = dayjs(date).hour(hour).minute(minute);

  return dateAndTime.$d;
}
```

**Comment by lightsound** [comment]
I would like that feature! 🙏

**Comment by lvchenyang** [comment]
I would like that feature!

**Comment by snilli** [comment]
I would like that feature! please implement them in someday. I will chagne my project to use this project. Thank you.

**Comment by sep2** [comment]
+1 vote here.

**Comment by bensont1** [comment]
Would love to see this component as well! very neat

**Comment by foogazy** [comment]
Any news on this?

**Reply by rtivital** [reply]
no, this feature is not planned to be released in near future

**Comment by derekbking** [comment]
Would love to see this feature added as well! I found a codepen that implements this as a custom component on an older version of Mantine. May be helpful as a starting place / design inspiration.
https://codesandbox.io/s/g8b7k?file=/src/App.js:103-117

**Comment by strblr** [comment]
This is a must have, honestly. Would love to see it in Mantine

**Comment by mindful-dev25** [comment]
Here's a date-time picker implementation on the latest Mantine version.
https://github.com/adam5909099/mantine-date-time-picker/blob/main/src/components/DateTimePicker.tsx

**Comment by Akash187** [comment]
Finally, we have a date-time-picker component as an npm package [Mantine Datetime Picker](https://www.npmjs.com/package/mantine-datetime-picker).

**Reply by romatou** [reply]
It is different from the [example](https://github.com/orgs/mantinedev/discussions/1058#discussion-3949005) mentioned above. Faced the need of a Date Time Picker component with available time slots fetched from Google Calendar but had no luck finding anything close. Strange that no body has created such thing, since appointments and services are all based on time slots availability.

**Comment by rajnijain6** [comment]
Is there any way to select multiple dates and time (For now with use of datePickerInput I can select dates only) as we have DatePickerInput in mantine?
![Uploading Screenshot 2023-11-02 at 2.46.16 PM.png…]()

**Comment by WenLonG12345** [comment]
Hi, thanks for the suggestion of DateTimePicker, is there anyway that DateTimePicker can be in range? My use case is to select both date & time with ranges. Do let me know if there is such an options. Really appreciate this!

**Reply by rtivital** [reply]
Currently, the library does not provide such component.

**Reply by RabeeAbuBaker** [reply]
> Currently, the library does not provide such component.

@rtivital Is it going to be available any time soon?
Also, it would be nice to allow providing custom hours. Something like this... 

<img width="340" alt="image" src="https://github.com/mantinedev/mantine/assets/38038348/eca22f75-6edc-4b4d-a470-cefb974d1c73">
